### PR TITLE
Add timing endpoint (timeline data) for stats

### PR DESCRIPTION
**Pull request by gadotroee**

### CODE REVIEW

File: agent/pkg/providers/stats_provider.go [hunk 1]
```diff
@@ -45,9 +45,15 @@ type AccumulativeStatsProtocol struct {
...
+	ProtocolsData []*AccumulativeStatsProtocol `json:"protocols"`
+	Time          int64                        `json:"timestamp"`
+}
+
 var (
-	generalStats = GeneralStats{}
-	bucketsStats = BucketStats{}
+	generalStats            = GeneralStats{}
+	bucketsStats            = BucketStats{}
+	internalBucketThreshold = time.Minute * 1
```
Comment (IgorGov) [inline agent/pkg/providers/stats_provider.go:56]: can it be const?

File: agent/pkg/providers/stats_provider.go [hunk 2]
```diff
@@ -113,6 +119,97 @@ func GetAccumulativeStats() []*AccumulativeStatsProtocol {
 	return finalResult
 }
 
+func GetAccumulativeStatsTiming(intervalSeconds int, numberOfBars int) []*AccumulativeStatsProtocolTime {
+	bucketStatsCopy := BucketStats{}
+	if err := copier.Copy(&bucketStatsCopy, bucketsStats); err != nil {
```
Comment (IgorGov) [inline agent/pkg/providers/stats_provider.go:124]: should this section be locked? to make sure no changed to the bucketsStats during the copy

Comment (gadotroee) [reply agent/pkg/providers/stats_provider.go]: I have changed those lines to be in function with https://github.com/up9inc/mizu/pull/1157/commits/5909395b053adc932ee67e2c4547b98bc77962a8 so both places that use the object will get it from this function

File: agent/pkg/providers/stats_provider.go (hunk 1 above)
Comment (gadotroee) [reply agent/pkg/providers/stats_provider.go]: Sure, changed in this https://github.com/up9inc/mizu/pull/1157/commits/5909395b053adc932ee67e2c4547b98bc77962a8

File: agent/pkg/providers/stats_provider.go [hunk 3]
```diff
@@ -113,6 +119,97 @@ func GetAccumulativeStats() []*AccumulativeStatsProtocol {
 	return finalResult
 }
 
+func GetAccumulativeStatsTiming(intervalSeconds int, numberOfBars int) []*AccumulativeStatsProtocolTime {
+	bucketStatsCopy := BucketStats{}
```
Comment (IgorGov) [inline agent/pkg/providers/stats_provider.go]: 1. private?
2. add 'copy' to its name?

**Review by IgorGov — APPROVED** [review]

**Pull request merged**